conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosGovernanceOperations.scala

```diff
@@ -0,0 +1,203 @@
+package tech.cryptonomic.conseil.indexer.tezos
+
+import tech.cryptonomic.conseil.common.tezos.TezosTypes._
+import com.typesafe.scalalogging.LazyLogging
+import slick.jdbc.PostgresProfile.api.Database
+import scala.concurrent.{ExecutionContext, Future}
+import cats.implicits._
+
+/** Process blocks and voting data to compute details for
+  * the governance-related cycles
+  */
+object TezosGovernanceOperations extends LazyLogging {
+
+  /** Collects all relevant data for the governance
+    * voting process
+    *
+    * @param hash identifies a specific block
+    * @param metadata the block metadata
+    * @param proposalId a specific proposal under evaluation
+    * @param allRolls ballot rolls up to this block for the proposal
+    * @param singleBlockRolls ballot rolls included in this specific block
+    * @param ballotsPerLevel how many ballots for the single block level
+    * @param ballotsPerCycle how many baloots for a whole cycle
+    */
+  case class GovernanceAggregate(
+      hash: BlockHash,
+      metadata: BlockHeaderMetadata,
+      proposalId: Option[ProtocolId],
+      allRolls: VoteRollsCounts,
+      rollsPerLevel: VoteRollsCounts,
+      ballotsPerCycle: Option[Voting.BallotCounts],
+      ballotsPerLevel: Option[Voting.BallotCounts]
+  )
+
+  //wrappings to make the distinction between two different counts
+  private case class BallotsPerCycle(counts: Voting.BallotCounts) extends AnyVal
+  private case class BallotsPerLevel(counts: Voting.BallotCounts) extends AnyVal
+
+  /** Collects rolls of votes counted during ballots
+    *
+    * @param yay how many yays
+    * @param nay how many nays
+    * @param pass how many passes
+    */
+  case class VoteRollsCounts(yay: Int, nay: Int, pass: Int)
+
+  /** We take basic governance data and the blocks storing that
+    * with the goal of deriving more detailed aggregate data on
+    * the voting process.
+    * The computation needs to access the database to fetch essential
+    * ballot data
+    *
+    * SAFETY NOTE: make sure that the blocks passed to this method have
+    *              have a proper metadata value, different from the genesis one
+    *
+    * @param db needed to fetch previously stored voting data
+    * @param proposalsBlocks blocks we want to have governance data for, associated to the protocol of the proposal
+    * @param proposalsBallots votes cast on the blocks of interest,
```

Comment:
Just like the name for blocks, this is inaccurate because of the ephemeral nature of this PR, whose goal is only to separate refactors in preparation of the actual fix, such that the changes will be easier to validate